### PR TITLE
[Merged by Bors] - chore: backports for leanprover/lean4#4814 (part 17)

### PR DESCRIPTION
**Pull request by kim-em**
see https://github.com/leanprover-community/mathlib4/pull/15245

### CODE REVIEW

File: Mathlib/Algebra/Quaternion.lean [hunk 1]
```diff
@@ -292,7 +292,8 @@ theorem mk_mul_mk (a₁ a₂ a₃ a₄ b₁ b₂ b₃ b₄ : R) :
 
 section
 
-variable [SMul S R] [SMul T R] (s : S)
+-- We temporarily remove the `CommRing R` instance.
+variable {R : Type*} {c₁ c₂ : R} {a : ℍ[R,c₁,c₂]} [SMul S R] [SMul T R] (s : S)
 
 -- Porting note: Lean 4 auto drops the unused `[Ring R]` argument
```
Comment (semorrison) [inline Mathlib/Algebra/Quaternion.lean:298]:
```suggestion
```

[the remark this answers is not in the document]
Comment (semorrison) [reply Mathlib/Algebra/Quaternion.lean]: I've cleaned up this file, with finer sections with appropriate typeclasses, and in particular removed this comment.

File: Mathlib/Algebra/Quaternion.lean [hunk 2]
```diff
@@ -292,7 +292,8 @@ theorem mk_mul_mk (a₁ a₂ a₃ a₄ b₁ b₂ b₃ b₄ : R) :
 
 section
 
-variable [SMul S R] [SMul T R] (s : S)
+-- We temporarily remove the `CommRing R` instance.
+variable {R : Type*} {c₁ c₂ : R} {a : ℍ[R,c₁,c₂]} [SMul S R] [SMul T R] (s : S)
```
[the remark this answers is not in the document]
Comment (semorrison) [reply Mathlib/Algebra/Quaternion.lean]: Solved by the refactor.

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**